pkg/networkmanager/dialogs-common.jsx

```diff
@@ -224,8 +246,13 @@ export const NetworkAction = ({ buttonText, iface, connectionSettings, type }) =
             dlg = <IpSettingsDialog topic="ipv4" {...properties} />;
         else if (type == 'ipv6')
             dlg = <IpSettingsDialog topic="ipv6" {...properties} />;
+
         if (dlg)
-            Dialogs.show(dlg);
+            if (type === 'wireguard') {
+                resolveDepsAndShowDialog(dlg);
+            } else {
+                Dialogs.show(dlg);
+            }
```

Comment:
See above, this is awkward. Please call `resolveDeps(type)` unconditionally, and then call the dialog on success.